crates/re_viewer/src/misc/caches/tensor_decode_cache.rs

```diff
@@ -0,0 +1,135 @@
+use re_log_types::component_types::{Tensor, TensorDimension, TensorId, TensorTrait};
+
+#[derive(thiserror::Error, Clone, Debug)]
+pub enum TensorDecodeError {
+    // TODO(jleibs): It would be nice to just transparently wrap
+    // `image::ImageError` and `tensor::TensorImageError` but neither implements
+    // `Clone`, which we need if we want to cache the Result.
+    #[error("Failed to decode bytes as tensor")]
+    CouldNotDecode,
+    #[error("Failed to interpret image as tensor")]
+    InvalidImage,
+    #[error("The encoded tensor did not match its metadata {expected:?} != {found:?}")]
+    InvalidMetaData {
+        expected: Vec<TensorDimension>,
+        found: Vec<TensorDimension>,
+    },
+}
+
+#[derive(Clone)]
+struct DecodedTensor {
+    /// `None` if the tensor was not successfully decoded
```

Comment:
```suggestion
    /// `Err` if the tensor was not successfully decoded
```

(but also kind of superfluous comment)